bolt-launcher: init at 0.9.0

nezia1 opened the pull request:
## Description of changes

Trying to implement Bolt, a third-party Runescape launcher. However, I need help on it as I'm missing a few things and I would like to be reviewed. Closes  #296491

## Things done
Added the package, linked the correct libraries, added cmake files to libcef as well as pre/postInstall hooks. The application builds fine, however I cannot get it to run and I need input on this. When trying to run it, strace gives me:
```
openat(AT_FDCWD, "/nix/store/qavag7d0q4bslf85q61ika50h2prvk0y-cef-binary-121.3.13/lib/icudtl.dat", O_RDONLY) = -1 ENOENT (No such file or directory)
```

I suspect that there might be some path manipulation to do, as it looks for icudtl.dat in the wrong directory, even though it should be correctly installed?

- Built on platform(s)
  - [x] x86_64-linux

Pandapip1 commented on pkgs/by-name/bo/bolt-launcher/package.nix:
```diff
@@ -0,0 +1,78 @@
+    addAutoPatchelfSearchPath ${libGL}/lib
+  '';
+
+  enableParallelBuilding = true;
```
Parallel building is enabled by default

Pandapip1 commented on pkgs/by-name/bo/bolt-launcher/package.nix:
```diff
@@ -0,0 +1,78 @@
+
+  enableParallelBuilding = true;
+
+  meta = with lib; {
```
`with lib;` is an anti-pattern

Pandapip1 commented on pkgs/by-name/bo/bolt-launcher/package.nix:
```diff
@@ -0,0 +1,78 @@
+  autoPatchelfHook,
+}:
+
+stdenv.mkDerivation rec {
```
Use `finalAttrs` instead of `rec`.

Pandapip1 commented on pkgs/by-name/bo/bolt-launcher/package.nix:
```diff
@@ -0,0 +1,78 @@
+  src = fetchFromGitHub {
+    owner = "AdamCake";
+    repo = "bolt";
+    rev = "${version}";
```
```suggestion
    rev = version;
```

or if you choose the `finalAttrs` route

```suggestion
    rev = finalAttrs.version;
```

Pandapip1 reviewed (changes requested):
Your commits are in the wrong format. See the [Commit Conventions](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions).

nezia1 committed:
maintainers: add nezia

nezia1 committed:
bolt-launcher: init at 0.9.0

Added bolt-launcher, an alternative launcher for Runescape 3 / Old
School Runescape.

https://github.com/Adamcake/Bolt/releases/tag/0.9.0

bolt-launcher: add plugin loader for rs3

Building the app with luajit, which allows for Runescape 3 plugin
loading.

bolt-launcher: add runescape 3 dependencies

Added Runescape 3 dependencies inside the buildFHSEnv, so that it can be
used by the binary (downloaded from the internet by bolt-launcher
itself, hence why the fhs env is needed here).

bolt-launcher: fix dependency issues

Added libbolt-plugin.so into $out/lib, which allows the program to use
the Runescape 3 plugin loader. Also updated the mainProgram as well as
runScript so that the program can be ran from nix run instead of just
being able to be ran manually.

Add feature flags for RS3 and HDOSAndRuneLite as well as .desktop and icon file support (#1)

* Add enable flags for RS3 and HDOSAndRuneLite

* Add .desktop and icon

* Fix formatting w/ nixfmt

* Remove enableHDOSAndRuneLite feature flag

* Fix formatting

Co-authored-by: Thomas King <thomas@tomking.io>

nezia1 commented:
Added a few extra dependencies inside of buildFHSEnv, so that it can be used by Runescape 3. Need input on this, as this is my first pull request and we had to do some odd stuff with the rpath, so I'm wondering if libcef itself should also be changed to be more extensible (ie. using autoPatchelfHook, I could not make it work using it so we had to override rpath in the install phase which is not great).

Pandapip1 commented:
Definitely, if you can get `autopatchelfHook` to work, that would be preferable, but it's not blocking. You should squash your subsequent changes to `package.nix` into a single commit though.

Pandapip1 commented:
You also shouldn't use any merge commits.

nezia1 commented:
Squashed everything and removed the merge commit as pointed out in your last message. I did not manage to make `autopatchElfhook`, I'm pretty sure there would need to be some changes made to it so that it's more easily extensible without overriding the whole install phase, but I also don't want to mess up a hard dependency to some other programs.

FliegendeWurst commented on pkgs/by-name/bo/bolt-launcher/package.nix:
```diff
@@ -0,0 +1,178 @@
+  runScript = "${bolt.name}";
+  meta = {
+    homepage = "https://github.com/Adamcake/Bolt";
+    description = "An alternative launcher for your favourite MMO";
```
```suggestion
    description = "An alternative launcher for RuneScape";
```

FliegendeWurst commented on pkgs/by-name/bo/bolt-launcher/package.nix:
```diff
@@ -0,0 +1,178 @@
+in
+buildFHSEnv {
+  name = bolt.name;
+
```
```suggestion
  version = bolt.version;
```

nezia1 replied:
Done. Also ended up using `inherit`, to make it a bit more concise.

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-already-reviewed/2617/2047

wolfgangwalther reviewed (commented):
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 338470`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 1 package marked as broken and skipped:</summary>
  <ul>
    <li>bolt-launcher</li>
  </ul>
</details>

nezia1 commented on pkgs/by-name/bo/bolt-launcher/package.nix:
```diff
@@ -0,0 +1,177 @@
+      libz
+      libcap
+      libsecret
+      openssl_1_1
```
Very unlikely.A lot of packages using `openssl_1_1` usually hard depend on it, which from my understanding and testing is also the case with `bolt-launcher`. I'd love to find a better solution but I think this might unfortunately be an issue with every package that depends on it.

Adamcake replied:
openssl 1.1 and libcrypto 1.1 are import dependencies of one of the games Bolt can launch. I made it an optdepend on AUR since not everyone wants to play that game. I did also notify Jagex that they need to update this and got an acknowledgement, but... well, don't hold your breath.

nezia1 replied:
> Could we put it behind a feature flag, so that the default will not be marked as insecure immediately?

I'm not sure if that would be reasonable. As @Adamcake mentioned, those insecure packages are unfortunately needed so that RS3 can run in the first place. I don't think providing a feature flag that breaks half the app when set to false would help a lot, leaving it as insecure would be better.

Besides, the proper RS3 package in nixpkgs is also broken because of it, as is cinny and a whole lot of other apps, so this should be probably okay.

tomking replied:
I imagine there are many people who want to install Bolt and are fine with it not being able to launch RS3. I think the feature flag approach makes sense, it would allow people to avoid needing insecure dependencies to use the launcher for other clients.

Obviously not up to me to make any decision on this -- but I took a stab at this here to use on my local if you are interested in pursuing this approach feel free to merge it in or to ignore if uninterested!

https://github.com/nezia1/nixpkgs/pull/1

nezia1 replied:
Thanks a lot @tomking ! We should be good to go now. We ended up adding an `enableRS3` flag that is set to false by default, as well as adding a desktop entry and an icon.
